### PR TITLE
Change locations and pre-defined activities to use a typeahead.

### PR DESCRIPTION
**Pull request by beshaya**
Closes: #645

### CODE REVIEW

**Comment by beshaya** [comment]
image

image

**Comment by blanchardjeremy** [comment]
Two thoughts: 

1) Why are the options underlined? Could we not underline them instead? I think it's unnecessary and visually distracting.

2) I have an idea. I'm not sold on the idea yet. But I wonder if it's wise for us to have labels above state/country & region?  I'm basing that on [this blog post](https://mono.company/design-practice/the-10-commandments-of-good-form-design-on-the-web/).

:)

**Comment by beshaya** [comment]
> Two thoughts:
> 
> 1. Why are the options underlined? Could we not underline them instead? I think it's unnecessary and visually distracting.
> 2. I have an idea. I'm not sold on the idea yet. But I wonder if it's wise for us to have labels above state/country & region?  I'm basing that on [this blog post](https://mono.company/design-practice/the-10-commandments-of-good-form-design-on-the-web/).
> 
> :)

Both done.

File: src/components/calculator/PrevalenceControls.tsx [hunk 1]
```diff
@@ -194,59 +199,79 @@ export const PrevalenceControls: React.FunctionComponent<{
...
+        return { label: Locations[locKey].label, value: locKey }
+      })
+  const selectedSubLocation = subLocationOptions.find(
+    (option) => option.value === data.subLocation,
+  )
+
   return (
     <React.Fragment>
       <header id="location">
         <Trans>calculator.location_selector_header</Trans>
```
Comment (blanchardjeremy) [inline src/components/calculator/PrevalenceControls.tsx:229]: this broke

image

**Comment by blanchardjeremy** [comment]
It's going to take a bit of work to merge this with #326. Any thoughts on which one we should do first?

**Comment by beshaya** [comment]
> It's going to take a bit of work to merge this with #326. Any thoughts on which one we should do first?

The merge doesn't look bad to me; I can handle rebasing in either direction. Merge when ready.

File: src/components/calculator/PrevalenceControls.tsx (hunk 1 above)
Comment (beshaya) [reply src/components/calculator/PrevalenceControls.tsx:229]: Fixed! Thanks for the catch!

**Comment by blanchardjeremy** [comment]
In that case, I suggest we merge #326 first (in the next few days) before merging this one.

**Comment by blanchardjeremy** [comment]
Or, if you're willing to do the re-base for the other ticket, I'm down to merge this one first. I just don't wanna make extra work for @david-mears-dfe. He's already done so much on that ticket.

**Pull request merged**